rocky/account/models.py

```diff
@@ -79,3 +84,25 @@ class KATUser(AbstractBaseUser, PermissionsMixin):
 
     def get_full_name(self):
         return self.full_name
+
+    @cached_property
+    def organization_members(self) -> List[OrganizationMember]:
+        """All organizations and organization members for a user. Cached to avoid multiple queries."""
+        return list(OrganizationMember.objects.filter(user=self).select_related("organization"))
+
+    @cached_property
+    def superuser_organizations(self) -> List[Organization]:
+        """All organizations of a superuser. Cached to avoid multiple queries."""
+        return Organization.objects.all()
+
+    def get_organizations(self) -> List[Organization]:
+        """Lists all organizations a user is a member of, excluding organizations to which access is blocked."""
+        if self.is_superuser:
+            return self.superuser_organizations
+        return [member.organization for member in self.organization_members if not member.blocked]
+
+    def get_organizations_including_blocked(self) -> List[Organization]:
+        """Lists all organizations a user is a member of, including organizations to which access is blocked."""
+        if self.is_superuser:
+            return self.superuser_organizations
+        return [member.organization for member in self.organization_members]
```

Comment:
I wonder if it's better to merge this into get_organizations with a parameter. Only downside is more limited usability from Django templates, as no arguments can be provided to function calls?

Reply:
Fixed